cmd/kube-controller-manager/app/core.go

```diff
@@ -256,7 +256,9 @@ func startResourceQuotaController(ctx ControllerContext) (bool, error) {
 		Registry:                  generic.NewRegistry(quotaConfiguration.Evaluators()),
 	}
 	if resourceQuotaControllerClient.CoreV1().RESTClient().GetRateLimiter() != nil {
-		metrics.RegisterMetricAndTrackRateLimiterUsage("resource_quota_controller", resourceQuotaControllerClient.CoreV1().RESTClient().GetRateLimiter())
+		if err := metrics.RegisterMetricAndTrackRateLimiterUsage("resource_quota_controller", resourceQuotaControllerClient.CoreV1().RESTClient().GetRateLimiter()); err != nil {
+			return false, err
```

Comment:
`return true, ....`

Reply:
Why `true` when some lines below is returning `false`? https://github.com/frodenas/kubernetes/blob/04d400c29ba2b5588ceb548473e60e0ad2ea7721/cmd/kube-controller-manager/app/core.go#L266

Also, this `bool` indicates if the controller has been [started](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L478), and it's failing before the `Run`, so it should return `false` right?

Reply:
> Also, this bool indicates if the controller has been started, and it's failing before the Run, so it should return false right?

The bool indicates that it was enabled.  Some controllers have secondary enablement criteria (like API availability) that they own inside their own start methods.